### DISCUSSION TITLE
Serverless Functions Example for Vue CLI

### DISCUSSION THREAD

**Discussion by ryanjwilke** [vercel/vercel#4022] [General] [answered]
Could you add some serverless function examples for VUE CLI? I'm having trouble getting the serverless functions working locally with `now dev` in combination with Vue Router. All the local "api" requests return a 404 locally, but in production they work just fine.

**Comment by styfle** [comment, marked as answer]
Hi @ryanjwilke 

There is a Vue example here: https://github.com/zeit/now/tree/master/examples/vue

You can add a `/api` directory with Serverless Functions.

Read more: https://zeit.co/docs/v2/serverless-functions/introduction

Can you provide an example that has a 404? Thanks!

**Reply by mcsdevv** [reply]
@ryanjwilke was the above of use to you?

**Reply by ryanjwilke** [reply]
Sorry for the delayed reply. Unfortunately it was not very useful no as it's missing concrete examples of serverless functions working in both production and the dev environments.

Fortunately, I was able to get it figured out through some trial and error. Essentially, in order for the `/api/` calls to work locally, I needed to add the routes in the `now.json` file as well. Here's an example:
```
{
  "routes": [
    {
      "src": "/api/posts",
      "dest": "/api/posts/index",
      "methods": ["GET"]
    },
    {
      "src": "/api/posts/([a-z-]*)",
      "dest": "/api/posts/id?slug=$1",
      "methods": ["GET"]
    }
  ]
}
```

As I mentioned, this all worked fine in the preview environments, but never worked locally until I added this file. Does that make sense?

**Reply by mcsdevv** [reply]
Hey Ryan, thanks for confirming, although there should be nothing you need to change to have Serverless Functions work in either environment - we'll take a look!

**Reply by ryanjwilke** [reply]
Sure thing. I'm happy to show you and via a video sharing session if it comes to that so I can show how it's all configured. Everything else worked generally pretty well, but that was the only thing that was quite difficult to get setup for some reason. Perhaps it's also due to using VueRouter along side of the `now.json` routes.

**Reply by mcsdevv** [reply]
Ah yes, VueRouter is likely to have an impact there, either way we'll have a look and see what can be done.

**Reply by ricky11** [reply]
is there a example of /api in a vue application, still could not find it.

**Reply by ghost** [reply]
I am also currently looking for serverless vue examples.
Currently the examples in the official documentation are similar to React.
Did you find it??

**Reply by amyegan** [reply]
Hi @Seung-hwan285! Serverless Functions can be written in Javascript/TypeScript, since [Node.js is a support language](https://vercel.com/docs/concepts/functions/serverless-functions/supported-languages), rather than in a framework-specific format. Examples in the [Serverless Function documentation](https://vercel.com/docs/concepts/functions/serverless-functions) typically show a version with Next.js  built-in types, and a more generic Node.js option for all other frameworks in a second tab.

![image](https://user-images.githubusercontent.com/8247105/205686897-d7e4d8ed-ff32-4bbf-b4e1-7dce0223bd26.png)

The [Nuxt on the Edge](https://vercel.com/templates/vue/nuxt-edge-functions) template has [an `/api` example](https://github.com/pi0/nuxt-on-the-edge/blob/main/server/api/info.ts) that might help if you want to see an example in the context of a repo 🙂

**Reply by ghost** [reply]
@amyegan  thank you  : )
